CustomLookup/CustomEntitySearch.cs

```diff
@@ -0,0 +1,187 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using System.Net.Http.Formatting;
+using System.Text;
+using Microsoft.AspNetCore.Routing.Template;
+
+namespace CustomLookup
+{
+    /// <summary>
+    /// Based on sample custom skill provided in Azure Search. Provided a user-defined list of entities
+    /// this function determines the entities first occurrence within a given document. This list of entities
+    /// must repeatedly be provided by the user for each document.
+    /// </summary>
+    public static class CustomEntitySearch
+    {
+        /// <summary>
+        /// We assert the following assumptions:
+        /// 1. All text files are Latin based (i.e. no languageCode)
+        /// 2. Words can contain special characters and numbers
+        /// 3. The provided entities are not case sensitive
+        /// </summary>
+        #region Classes used to deserialize the request
+        private class InputRecord
+        {
+            public class InputRecordData
+            {
+                public string Text { get; set; }
+                public List<string> Words { get; set; }
+            }
+
+            public string RecordId { get; set; }
+            public InputRecordData Data { get; set; }
+        }
+
+        private class WebApiRequest
```

Comment:
>class [](start = 16, length = 5)

If you end up keeping these models (rather than using common util models) please move each class to its own file